rfcs/007 - Alternative Ways to Create Mappers.md

```diff
@@ -0,0 +1,122 @@
+# 007 - Alternative Ways to Create Mappers
+
+## Current Status
+
+### Proposed
+
+2023-02-07
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- Full Name <email@example.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Mohammad Teimori Pabandi <m@iasql.com>
+
+## Summary
+
+Having Pure-SQL modules would help us expand more. One key advantage is that we don't need a PR on our engine to add coverage for a new service, we just use a syntax like this and everything JustWorks™.
+```postgresql
+SELECT iasql_install('https://github.com/someModuleDeveloper/awsEc2Module');
+```
+That's where we started, but it might be useful to assess different paths we might take for creation of an IaSQL module/mapper.
```

Comment:
It would be good for others and for future-us to include all of the reasons why we're considering this.

1. Enables 3rd Party Modules for an actual user ecosystem to exist.
2. Reduces pressure on us to cover 100% of all AWS services before getting our first customer, because users can customize their modules to their own needs.
3. Potentially improves development/testing turn-around times by segmenting the engine testing from module testing (changes to a module would only trigger tests within the module's repo, changes to the engine would still trigger all officially-supported modules for testing, but would potentially be done much less often, and primarily for confirming newer version(s) of the AWS SDK still work with our modules.
4. Adding support for other clouds/SaaSes would be simpler within the engine.